File: go/300.600/371.SumOfTwoIntegers.go
package main

import "fmt"

func getSum(a int, b int) int {
	for b != 0 {
		a, b = a^b, (a&b)<<1
	}
	return a
}

func main() {
	fmt.Println(getSum(1, 2), 3)
	fmt.Println(getSum(-2, 3), 1)
	fmt.Println(getSum(2, -3), -1)
	fmt.Println(getSum(-2, -80), -82)
}

File: 300.500/371.SumOfTwoIntegers.cpp
#include <iostream>
using namespace std;

class SumOfTwoIntegers
{
public:
    int getSum(int a, int b) {
        while (b) {
            unsigned int x = (unsigned int)(a & b) << 1;
            a ^= b;
            b = x;
        }
        return a;
    }
};

int main(int argc, char const* argv[]) {
    SumOfTwoIntegers cls;
    cout << cls.getSum(1, 2) << " -> 3" << endl;
    cout << cls.getSum(-2, 3) << " -> 1" << endl;
    cout << cls.getSum(-3, 2) << " -> -1" << endl;
    cout << cls.getSum(-80, -2) << " -> -82" << endl;
    return 0;
}
